syncano/models/custom_sockets.py

```diff
@@ -0,0 +1,155 @@
+# -*- coding: utf-8 -*-
+from syncano.exceptions import SyncanoValueError
+from syncano.models.custom_sockets_utils import DependencyMetadataMixin, EndpointMetadataMixin
+
+from . import fields
+from .base import Instance, Model
+
+
+class CustomSocket(EndpointMetadataMixin, DependencyMetadataMixin, Model):
+    """
+    OO wrapper around instance custom sockets.
+    Look at the custom socket documentation for more details.
+
+    :ivar name: :class:`~syncano.models.fields.StringField`
+    :ivar endpoints: :class:`~syncano.models.fields.JSONField`
+    :ivar dependencies: :class:`~syncano.models.fields.JSONField`
+    :ivar metadata: :class:`~syncano.models.fields.JSONField`
+    :ivar status: :class:`~syncano.models.fields.StringField`
+    :ivar status_info: :class:`~syncano.models.fields.StringField`
+    :ivar links: :class:`~syncano.models.fields.LinksField`
+    """
+
+    name = fields.StringField(max_length=64, primary_key=True)
+    endpoints = fields.JSONField()
+    dependencies = fields.JSONField()
+    metadata = fields.JSONField(read_only=True, required=False)
+    status = fields.StringField(read_only=True, required=False)
+    status_info = fields.StringField(read_only=True, required=False)
+    links = fields.LinksField()
+
+    class Meta:
+        parent = Instance
+        endpoints = {
+            'detail': {
+                'methods': ['get', 'put', 'patch', 'delete'],
+                'path': '/sockets/{name}/',
+            },
+            'list': {
+                'methods': ['post', 'get'],
+                'path': '/sockets/',
+            }
+        }
+
+    def get_endpoints(self):
+        endpoints_path = self.links.endpoints
+        connection = self._get_connection()
+        response = connection.request('GET', endpoints_path)
+        endpoints = []
+        for endpoint in response['objects']:
+            endpoints.append(SocketEndpoint(**endpoint))
+        return endpoints
+
+    def run(self, method, endpoint_name, data={}):
```

Comment:
`def run(self, endpoint_name, method='GET', data=None):`
is better as method=GET will be probably the most common anyway.
Also, don't use {} as a default value. You will end up having a mutable default value that will be shared once changed.
Use None and then simply
`return endpoint.run(method, data=data or {})`